Add support for Carthage

sozorogami opened the pull request:
In order to support Carthage in `TMTumblrSDK`, this library and all of its
dependencies must have a shared framework scheme.

  - Add Carthage support to all dependencies.
  - Remove existing static lib target from project file.
  - Add a framework target and add all relevant source files.
  - Set the scheme for building this target to "shared".
  - Add all public headers to the framework's umbrella header.
  - Tag this commit as a new release, or Carthage will never look at it.

sozorogami commented:
Merge tumblr/JXHTTP#2 and tumblr/Spectacles#8 first.

sozorogami commented:
@brianmichel

brianmichel commented:
Ohhhhh! Very nice!

Pearapps commented on TMTumblrSDK/Core/TMSDKFunctions.h:
```diff
@@ -6,6 +6,7 @@
 //  Copyright (c) 2013 Tumblr. All rights reserved.
 //
 
+@import Foundation;
```
A lot of these @imports duplicate the other imports, and will @import break when this is compiled into a static lib?

segiddins replied:
No, it will only break if `Enable Modules` is set to `NO`.

sozorogami replied:
Fixed the dupe imports.

`@import` shouldn't break, because Apple frameworks (Foundation, UIKit etc.) are always available as modules. `pod install` worked fine when I tested it.

Pearapps commented on Cartfile.resolved:
```diff
@@ -0,0 +1,2 @@
+github "sozorogami/JXHTTP" "2.0.3"
```
This is odd

sozorogami replied:
Crap. Changed the Cartfile to not be pointing at my fork, but didn't `carthage update` after. I'll fix this.

sozorogami commented:
Addressed some feedback from @Pearapps 
@brianmichel This should be good to go now. To test, make sure a project using TMTumblrSDK can get it via both Cocoapods and Carthage.

brianmichel commented:
Taking a look right now!

brianmichel commented:
This works for me in both Cocoapods static/dynamic, and Carthage. Thanks!
